data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -96,6 +105,10 @@ protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSc
     this.posAccessor = requiredSchema.accessorForField(MetadataColumns.ROW_POSITION.fieldId());
   }
 
+  protected DeleteFilter(String filePath, List<DeleteFile> deletes, Schema tableSchema, Schema requestedSchema) {
```

Comment:
Is this just for legacy compatibility? Do we have any users of this here? as a protected method I would think we would be free to drop it if we wanted to.

Reply:
This constructor without the `DeleteCounter` is called by `GenericDeleteFilter` and `FlinkDeleteFilter`.